### PR TITLE
1.5 (sh) feat : Add unit test for FollowStateTrackerServiceImpl.getFS…

### PR DESCRIPTION
**Pull request by Samuel-Huguet**
…TByFollowerID(Long followerID)

### CODE REVIEW

File: skillful_network_server/src/test/java/fr/uca/cdr/skillful_network/services/user/FollowStateTrackerServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        FollowStateTracker fst = new FollowStateTracker(userFollowed,userFollower);
+        fstList.add(fst);
+        fstSet.add(fst);
+
+        userFollower.setFollowerSet(fstSet);
+        userFollowed.setFollowableSet(fstSet);
+
+        Mockito.when(followStateTrackerRepository.findAllByFollower(userFollower))
+                .thenReturn(fstList);
+        Mockito.when(userService.getById(userFollower.getId())).thenReturn(userFollower);
```
Comment (Samuel-Huguet) [inline skillful_network_server/src/test/java/fr/uca/cdr/skillful_network/services/user/FollowStateTrackerServiceTest.java]: These lines aren't necessary, are they?
I think the Mockito.DoAnswer methods just below do the same job but I'm not 100% sure.
WDYT ? @danglotb @pfouque06

Comment (danglotb) [reply skillful_network_server/src/test/java/fr/uca/cdr/skillful_network/services/user/FollowStateTrackerServiceTest.java]: Imo, you are right. It worth a try. If you comment these lines and rerun the test, do you observe the same behavior ? Does the test still pass ? Maybe your assertions aren’t strong enough but it is no big deal here.
I agree that we can remove them (a priori)

**Review by danglotb — COMMENTED** [review]
LGTM, but kind of messy, right ? Do you plan to work a bit more on this test ? Great job anyway !

File: skillful_network_server/src/test/java/fr/uca/cdr/skillful_network/services/user/FollowStateTrackerServiceTest.java (hunk 1 above)
Comment (Samuel-Huguet) [reply skillful_network_server/src/test/java/fr/uca/cdr/skillful_network/services/user/FollowStateTrackerServiceTest.java]: If I comment these lines, I observe the same behavior and the test still pass. So I think I can remove these lines

**Comment by Samuel-Huguet** [comment]
> LGTM, but kind of messy, right ? Do you plan to work a bit more on this test ? Great job anyway !

This is the first test, there are still 6 tests to add to this file. I wanted you to see the first one to be sure that there are no logical problems because the others (test) will follow, with a few small changes, the same logic.

File: skillful_network_server/src/test/java/fr/uca/cdr/skillful_network/services/user/FollowStateTrackerServiceTest.java (hunk 1 above)
Comment (Samuel-Huguet) [reply skillful_network_server/src/test/java/fr/uca/cdr/skillful_network/services/user/FollowStateTrackerServiceTest.java]: Concerning assertions, what do you suggest to add ? 
I was thinking to add : 
- the list's size is equal to 1;
- Maybe add names to the users and compare that they are the same at the end in the method response ?

**Pull request merged**